src/test/java/com/gizasystems/automationexercise/tests/categoryProducts/ViewCategoryProductsTests.java

```diff
@@ -0,0 +1,47 @@
+package com.gizasystems.automationexercise.tests.categoryProducts;
+
+import com.gizasystems.automationexercise.pages.*;
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Description;
+import io.qameta.allure.Epic;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Epic("Automation Exercise")
+@Feature("Category Products")
+@Story("View Category Products")
+public class ViewCategoryProductsTests {
+    private SHAFT.GUI.WebDriver driver;
+    private SHAFT.TestData.JSON testData;
+
+    @Test(description = "View Category Products")
+    @Description("Given that user want to view main Category and subcategory Products, When user click on main Category and select subcategory, Then user should be navigated to that subcategory page")
+    public void viewCategoryProductsTestGui() {
+        new CategoriesBar(driver).clickOnCategoryLink(testData.getTestData("MainCategory.womanCategory")).clickOnSubCategoryLink(testData.getTestData("MainCategory.womanCategory"), testData.getTestData("SubCategories.womanSubCategory"));
+        new HomePage(driver).validateOnVisibilityOfCategoryTitle(testData.getTestData("womanCategoryTitle"));
+        new CategoriesBar(driver).clickOnCategoryLink(testData.getTestData("MainCategory.menCategory")).clickOnSubCategoryLink(testData.getTestData("MainCategory.menCategory"), testData.getTestData("SubCategories.menSubCategory"));
+        new HomePage(driver).validateOnVisibilityOfCategoryTitle(testData.getTestData("menCategoryTitle"));
```

Comment:
Please take a new line between the methods to make it more readability and showed in the demo test case found in the Register User classes.